### PR TITLE
Add a top level vars file and add some defaults to roles

### PR DESCRIPTION
**Pull request by rthallisey**
Fixes: #115

### CODE REVIEW

File: playbooks/cluster/openshift/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# OpenShift Cluster Role
```
Comment (lukas-bednar) [inline playbooks/cluster/openshift/README.md]: s/Role/Playbook/ it seems to be playbook to me :-)

File: playbooks/provider/lago/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Logo Provider Role
```
Comment (lukas-bednar) [inline playbooks/provider/lago/README.md]: s/Logo/Lago/, and s/Role/Playbook/

File: automation/check-patch.sh [hunk 3]
```diff
@@ -145,9 +145,9 @@ main() {
         "mode=$mode"
         "provider=$provider"
         "inventory_file=$inventory_file"
-        "cluster_type=$cluster_type"
+        "cluster=$cluster"
         "ansible_modules_version=$ansible_modules_version"
-        "openshift_ver=$openshift_ver"
+        "openshift_version=$openshift_version"
```
Comment (gbenhaim) [inline automation/check-patch.sh]: `openshift_version` is also being used in `openshift-ansible`. I think that we shouldn't use it also.

File: playbooks/provider/lago/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+cluster: "openshift"
```
Comment (gbenhaim) [inline playbooks/provider/lago/defaults/main.yml]: Except for `inventory`, all the other variables are not being used by the Lago provider.
Why do we need this file?

File: vars/all.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+### Cluster ###
```
Comment (gbenhaim) [inline vars/all.yml]: I agree that this file is needed, but:

- It should be near `kubevirt.yml`, which is the playbook that will use it.
- We should include this file using `include_vars` from `kubevirt.yml`.
- This file shouldn't contain any variables that are not related to Kubevirt's deployment. You can place global 
  variables in `vars/global_vars.yml`, and I think that we shouldn't use this file often.

Comment (rthallisey) [reply vars/all.yml]: The goal is for this file to be the top level interface for kubevirt-ansible, not just the kubevirt playbook.  Variables declared here like openshift_version and cluster will be used in multiple playbooks.  This should absorb ```vars/global_vars.yml```.

We can use ```include_vars``` in the playbook so we don't have to use ```-e@vars/all.yml```.

File: vars/all.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+cluster: openshift
+namespace: kube-system
+
+# OpenShift #
+openshift_version
```
Comment (rthallisey) [inline vars/all.yml]: /me need to add a value for openshift_version

File: automation/check-patch.sh (hunk 3 above)
Comment (rthallisey) [reply automation/check-patch.sh]: Let's keep it so we expose it to the top level

**Comment by rthallisey** [comment]
@lukas-bednar @gbenhaim any more comments? Would like to get this merged asap

File: playbooks/cluster/openshift/config.yml [hunk 7]
```diff
@@ -1,13 +1,15 @@
 ---
 - hosts: all
   remote_user: root
+  vars:
+    epel_release_rpm_url: https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
```
Comment (alexxa) [inline playbooks/cluster/openshift/config.yml]: @rthallisey I'm not an Ansible expert, and I don't follow why it's here if there is the same var in playbooks/cluster/openshift/defaults/main.yaml ?

Comment (rthallisey) [reply playbooks/cluster/openshift/config.yml]: Something I noticed in CI was that the variables in defaults are not being used.  But when I run the playbook locally they do.  So to answer your first question, I tried changing the name from yml to yaml, but it didn't work.  Finally I added the variable here for now to see if CI will go back to green and it did.

I'll investigate this some more tomorrow.
